src/igraph/__init__.py

```diff
@@ -2814,12 +2814,32 @@ def Incidence(klass, *args, **kwds):
           what the value is. If C{True}, non-zero entries are rounded up to
           the nearest integer and this will be the number of multiple edges
           created.
+        @param weighted: defines whether to create a weighted graph from the
+          incidence matrix. If it is c{None} then an unweighted graph is created
+          and the multiple argument is used to determine the edges of the graph.
+          If it is a character constant then for every non-zero matrix entry, an
+          edge is created and the value of the entry is added as an edge attribute
+          named by the weighted argument. If it is C{True} then a weighted graph
+          is created and the name of the edge attribute will be ‘weight’.
 
         @return: the graph with a binary vertex attribute named C{"type"} that
           stores the vertex classes.
         """
+        weighted = kwds.pop("weighted", False)
+        if weighted:
+          kwds["multiple"] = False
         result, types = klass._Incidence(*args, **kwds)
         result.vs["type"] = types
+        if weighted:
+            weight_attr = "weight" if weighted == True else weighted
```

Comment:
We are not being entirely consistent here. If `weighted` is set to an empty string, we treat it as if it was `False`, contrary to what the documentation says.

I would do a more thorough type checking first on `weighted`; basically we need to ensure first that `weighted` can only be one of `True`, `False` or a string. Once we've established that, we need to treat the empty string as any other string.

Reply:
Can graph allow empty string property in edges?

Reply:
In theory it does; the attribute storage is simply a Python dict where the keys are enforced to be strings, but AFAIK there are no other constraints.